selfdrive/car/tests/test_models.py

```diff
@@ -345,6 +346,11 @@ def test_panda_safety_carstate(self):
       checks['brakePressed'] += brake_pressed != self.safety.get_brake_pressed_prev()
       checks['regenBraking'] += CS.regenBraking != self.safety.get_regen_braking_prev()
 
+      # Verify that panda has the correct velocity for cars that use it (angle based cars)
+      if self.CP.steerControlType in [car.CarParams.SteerControlType.angle]:
+        panda_velocity = (self.safety.get_vehicle_speed_min() + self.safety.get_vehicle_speed_max()) / 2 / VEHICLE_SPEED_FACTOR
+        checks['vEgo'] += abs(CS.vEgo - panda_velocity) > 3
```

Comment:
should also check the latest matches exactly

Reply:
any point in checking average? I'll just check last